tests/main/views/test_direct_award.py

```diff
@@ -659,30 +663,71 @@ def _lock_project_data(self):
             'updated_by': str(self.user_id)
         }.copy()
 
-    @pytest.mark.skip
     def test_lock_project_400s_if_project_already_locked(self):
-        pass
+        with self.app.app_context():
+            project = DirectAwardProject.query.get(self.project_id)
+            project.locked_at = datetime.utcnow()
 
-    @pytest.mark.skip
-    def test_lock_project_404s_if_invalid_project(self):
-        pass
+            db.session.add(project)
+            db.session.commit()
 
-    @pytest.mark.skip
-    def test_lock_project_search_and_project_datetimes_are_updated(self):
-        pass
+        res = self.client.post(
+            '/direct-award/projects/{}/lock'.format(self.project_id),
+            data=json.dumps({
+                'updated_by': 'example',
+            }),
+            content_type='application/json')
 
-    @pytest.mark.skip
-    def test_lock_project_search_result_entries_populated_with_latest_archived_service_for_searched_services(self):
-        pass
+        assert res.status_code == 400
+
+    def test_lock_project_404s_if_invalid_project(self):
+        res = self.client.post(
+            '/direct-award/projects/{}/lock'.format("not_a_valid_id"),
+            data=json.dumps({
+                'updated_by': 'example',
+            }),
+            content_type='application/json')
+
+        assert res.status_code == 404
 
-    @pytest.mark.skip
-    def test_lock_project_saved_result_entries_are_for_the_active_search_in_the_project(self):
-        pass
+    @mock.patch('app.main.views.direct_award.search_api_client')
+    def test_lock_project_success(self, search_api_client):
```

Comment:
The name of this test ('success') is pretty generic. It doesn't help someone reading it understand what, specifically, is being tested.

Reply:
I think this is ok - there's a single path which is "all the things that happen when the user successfully locks a project". We don't want to rename the function whenever we find more side-effects to test, and we didn't want separate test functions because we'd end up testing the same code over and over.

If there were more pathways that were 'successful' then yes we'd need to distinguish. And particular failure modes need to be in separate tests too, for sure.